lib/internal/webstreams/util.js

```diff
@@ -129,24 +129,15 @@ function transferArrayBuffer(buffer) {
   return res;
 }
 
-function isArrayBufferDetached(buffer) {
-  if (ArrayBufferGetByteLength(buffer) === 0) {
```

Comment:
Thought about what else we can do for the webstreams-specific `validateArrayBuffer()`.

- For performance, it will be better to pre-check both `viewByteLength` and `viewBufferByteLength` are `0`.
- Except for the pre-checking, using `new Uint8Array()` in a try-catch block seems the optimal idea, as of now, to verify if a buffer is detached.
- There seems no exact C++ API we need. One reference found is [`napi_is_detached_arraybuffer`](https://github.com/nodejs/node/blob/5fad0b93667ffc6e4def52996b9529ac99b26319/doc/api/n-api.md#napi_is_detached_arraybuffer). However, it's doubtful that it works well even when the size of a buffer is small.

So my thought is just reverting the last commit. wdyt?

Reply:
In a context of this PR, yes, I think either reverting the last commit or direct follow-up with rewriting that into validator function (throws on invalid input, returns void) is good.

Depending on usual usage, it might have a check that input is `ArrayBuffer` instance.

As for check for detach itself, in current design it might be reasonable to tighten it a bit, i.e. at least check that thrown error was `TypeError`.
To improve performance, perhaps it would be possible to have a fast builtin way to check that `[[Data]]` is null later at some point.

Reply:
Applied suggestions, PTAL.